### DISCUSSION TITLE
Optimal Slice of 3D space

### DISCUSSION THREAD

**Discussion by domjones93** [meta-pytorch/botorch#1340] [Q&A] [answered]
Hi,

I am currently looking into utilizing botorch for finding the optimal "slice" of a 3d geometry for improving the estimate of the 3D data.  I will be physically probing a volume from the top surface, and use BO to suggest the next probe location and rotation.

My probe data and GP model will be in the form of an XYZ grid of data, and the training data will be an array of points on the sensing plane of the probe.

I am able to locate the optimal probing point within the volume with a standard Bayesian Optimization Loop (model:FixedNoiseGP, acquisition: EI), but am unsure on how to optimize the planar rotation.

![text7699-7](https://user-images.githubusercontent.com/45069811/182890518-6f51cb77-0139-4e56-a7cf-bd92959dbb1d.png)

Could anyone suggest how I may be able to achieve this?

Many Thanks!

**Comment by Balandat** [comment, marked as answer]
Hmm not sure I fully understand, but is your problem that you're measuring `(x,y,z)` values (cartesian coordinates) but what you're controlling (and optimizing over) is `(x,y,phi)` where there is a rotation angle? What is your utility function here? Seems like there would need to be some aggregation across `(x,y,z)` observations on the sensing plane? 

What I'm trying to figure out is whether this is about how to formulate the problem or more a trigonometry or modeling question.

**Reply by domjones93** [reply]
Thanks for the quick reply! 

Apologies for being unclear, I am still relatively new to Bayesian optimization.

Yes, I am sampling data at Cartesian coordinates, with an array of sensing points fixed to a plane.  Rather than optimizing the system to find a single maximum, I am hoping to maximize the whole array of sensors (which I describe as a plane above, as they are controlled by a central position and a rotation).

 I am also unsure of how I should output to a different coordinate system (outputting x,y,phi when inputting x,y,z).

I currently have the geometric transform  and can maximize the mean expected improvement of the probe array, but am stuck when it comes to including the rotation in the optimization step.

**Reply by Balandat** [reply]
What exactly you mean by "mean expected improvement"? Are you computing EI on the point-level or are you computing EI on the aggregate value across the plane for a fixed angle? 

In the BO loop, can you choose individual probing points? Or do you need to choose points on a plane? 

When you compute your value across the plane, is there a discrete set of points or are you integrating across the plane? 

If the problem is such that you can select individual x/y/z coordinates to test during the loop, but then want to maximize the aggregate quality across some fixed plane, this is a pretty interesting setting. I haven't really come across this before and would need to think some more, but essentially your objective looks like `max_{x,y,phi} u(x,y,phi))`, where `u(x,y,phi) = \int_{P(x,y,phi)} u(p) dp`, where `u(p)` is the utility at location `p \in P(x,y,phi)` with `P(x,y,phi)` describing the plane defined by `x,y,phi`.

**Reply by domjones93** [reply]
Thank you so much!  I evaluated the utility discretely as the sensing locations remain constant in the plane's frame.  Aggregating the utility across these discrete points for each and maximising over x, y, and phi has worked perfectly.
